### PR TITLE
Adding mirroring script

### PR DESCRIPTION
**Pull request by davidferlay**

### CODE REVIEW

File: scripts/mirroring/.gitlab-ci.mirroring_exemple.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Exemple of .gitlab-ci.yml jobs
```
Comment (andypost) [inline scripts/mirroring/.gitlab-ci.mirroring_exemple.yml:1]: Typo in file name - `s/exemple/example`

File: scripts/mirroring/.gitlab-ci.mirroring_example.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Example of .gitlab-ci.yml jobs
+# Use with scripts/mirroring/mirror_current_branch.sh
+
+stages:
+  - mirror
+
+.mirroring_template: &mirroring_template
+  stage: mirror
+  dependencies:
```
Comment (gaydabura) [inline scripts/mirroring/.gitlab-ci.mirroring_example.yml:9]: Please describe this part in Read.me

File: scripts/mirroring/.gitlab-ci.mirroring_example.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  allow_failure: true
+  retry:
+    max: 2
+
+Mirror to repo XXX:
+  <<: *mirroring_template
+  script:
+  - echo " - Start of CI script"
+  - date; pwd; ls -lah;
+  - ls -lah ~
```
Comment (gaydabura) [inline scripts/mirroring/.gitlab-ci.mirroring_example.yml:21]: `ls` is duplicated

File: scripts/mirroring/.gitlab-ci.mirroring_example.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  - mkdir -p ~/.ssh
+  - ls -lah ~
+  - ls -lah ~/.ssh
+  - echo "$REMOTE_REPO_PRIVATE_KEY" > ~/.ssh/id_rsa
+  - chmod 0600 ~/.ssh/id_rsa
+  - ls -lah ~/.ssh
```
Comment (gaydabura) [inline scripts/mirroring/.gitlab-ci.mirroring_example.yml:27]: suggestion: replace duplicated `ls` after every operation with some summary info before run sh script.

File: scripts/mirroring/mirror_current_branch.sh [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+# Copying files to mirrorring dir
+echo -e "- Copying files to mirrorring dir...\n"
+rsync -av --progress ../. . --exclude .git/
+
+
+# Making sure everything needed will be included in commit
+echo -e "- Making sure everything needed will be included in commit...\n"
+mv .gitignore .gitig
+echo -e "- Deleting all .gitignore files except root one...\n"
+find . -name '.gitignore' -type f | wc -l
```
Comment (gaydabura) [inline scripts/mirroring/mirror_current_branch.sh:92]: duplicated lines which inform you about files before/after delete op. i think it was for debug purpose ?

File: scripts/mirroring/README.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+2 files are required : 
+- `.gitlab-ci.yml`
+- `mirror_current_branch.sh`
+
+1. Define a mirroring CI job like "Mirror to repo XXX" in .gitlab-ci.yml, as shown in .gitlab-ci.mirroring_example.yml
+1. In Gitlab UI, add the following custom CI/CD variables :
+   - REMOTE_REPO_IP : IP or domain name of target git repos
+   - REMOTE_REPO_PRIVATE_KEY : SSH private key matching public key added to git user
+   - REMOTE_REPO_TYPE : Possible values : "PLATFORM.SH" or nothing
```
Comment (gaydabura) [inline scripts/mirroring/README.md:26]: `Possible values : "PLATFORM.SH" or nothing`
->
`Possible values : "PLATFORM.SH" only for now.`

**Comment by davidferlay** [comment]
Thanks for review guys) gonna work on that in the next days

File: scripts/mirroring/.gitlab-ci.mirroring_example.yml (hunk 2 above)
Comment (davidferlay) [reply scripts/mirroring/.gitlab-ci.mirroring_example.yml:9]: Added https://github.com/skilld-labs/skilld-docker-container/pull/239/files#diff-b21723e86f3dfcd0b209fd27e65791e5R23

File: scripts/mirroring/mirror_current_branch.sh (hunk 5 above)
Comment (davidferlay) [reply scripts/mirroring/mirror_current_branch.sh:92]: Good point, fixed

**Pull request merged**